### ISSUE TITLE
Inconsistent `projectID` variable names

### ISSUE THREAD

**Issue by steffilazerte** [MotusDev/Motus-TO-DO#1304]
As with the [last issue](#1300) I'm working to update the article on [Database Variables](https://motuswts.github.io/motus/articles/database-variables.html) and through this I'm finding more inconsistencies in variable names in the package as well as coming from the API.

However, I'm not sure if they are truly inconsistencies, or are required for some reason or another. 

I have a couple of different variables to ask about, but I thought I'd start with `projectID`. Understanding the naming here will help me understand whether there's actually a problem with some of the others or not.

Here are the issues I've found with the `projectID` variable. 

❓ maybe inconsistent, ✔️ looks good to me, ❌ inconsistent

 Name in Views | Name in Table/API | From Table (API) | Project ID for...
------------- | ---------------------- | --------------------| ------------------
❓ `recvProjID` |	✔️ `projectID`   | `recvDeps`<br>(recv_metadata_for_projects,<br>metadata_for_receivers) | Receiver
 ❓ `tagProjID`  |  ✔️ `projectID`        | `tags`/`tagDeps`<br>(tag_metadata_for_projects,<br>metadata_for_tags)  |    Tag
 --             |❓  `ambigProjectID`  | `tagAmbig`<br>(tags_for_ambiguities)  |     Ambig tag
 --              | ❓ `ambigProjectID`  | `projAmbig`<br>(project_ambiguities_for_tag_project)  |    Ambig tag
 --            | ❌ `motusProjectID`  | `batches`<br>(batches_for_tag_project,<br>batches_for_receiver) |   Tag (hit)
--                     | ❌ `motusProjID`  | `filters` (Created locally) | Tag (hit)
 --                   |  ✔️ `projectID1` (etc.)  | `projAmbig` (API) | Tag (Ambig?)
   --                    | ❌ `tagDepProjectID` | `projBatch` (Created locally) | Tag (hit)

### Questions
@denislepage, @cjardine-bsc, @ajmacdonald3, @j-sayers, @leberrigan, @StuMackenzie, etc.)
1. Why do some have `motus` prepended (e.g., `motusProjectID`, `motusProjID`)? Couldn't those remain `projectID`?
2. Why is there a `tagDepProjectID`? Could this not be `projectID` or `tagProjID` if required?
3. Although we may not want to do this, would it not be better to have:
    - `recvProjID` (or `recvProjectID`) in Views
    - `tagProjID` (or `tagProjectID`) in Views
    - `ambigProjID` (or `ambigProjectID`) in Tables/API
    - `projectID` in Tables/API
       - (instead of `motusProjectID` or `motusProjID` or `tagDepProjectID`)
4. If I've completely misunderstood, please help me!

> **If the answer to Question 3 is yes, is this something we want to do?**
>   - **Yes**, because being consistent with names can make it much easier for users to understand that database structure (which is something we've been trying to implement)
>   - **No**, because changing things can introduce errors to the package or break existing user code.

**Comment by ajmacdonald3** [comment]
I would lean toward consistency since it'll be better in the long run even if does break some users' code. I wonder if we shouldn't always have just `tagProjID` or `recvProjID`? That way there wouldn't ever be confusion as to whether other variables like `motusProjID` or `projectID` refer to a tag's project or a receiver's project? I think those always apply to a tag's project but that may not be immediately clear to users.

**Comment by j-sayers** [comment]
I definitely agree about consistency, though it will certainly break a lot of code! But if we clearly inform of the changes and provide mappings that would help people update their code quickly with a find-and-replace (RStudio, VS Code, Notepad++ all provide options to do this in multiple files at the same time).

**Comment by cjardine-bsc** [comment]
Thanks Steffi for diving in and getting the organized. I think this is a finicky but necessary exercise. Thanks for taking it on. I agree with what's been said above. I've included my suggested changes in the table below. Do these make sense to you? 

<html><body>
<p dir="auto">Here are the issues I've found with the <code class="notranslate">projectID</code> variable.</p>
<p dir="auto"><g-emoji class="g-emoji" alias="question" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/2753.png">❓</g-emoji> maybe inconsistent, <g-emoji class="g-emoji" alias="heavy_check_mark" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/2714.png">✔️</g-emoji> looks good to me, <g-emoji class="g-emoji" alias="x" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/274c.png">❌</g-emoji> inconsistent</p>


Name in Views | Name in Table/API | From Table (API) | Project ID for...
-- | -- | -- | --
✔️ recvProjID | ❌ projectID change to recvProjID | recvDeps(recv_metadata_for_projects,metadata_for_receivers) | Receiver
✔️ tagProjID | ❌ projectID change to tagProjID | tags/tagDeps(tag_metadata_for_projects,metadata_for_tags) | Tag
-- |  ❌ ambigProjectID change to ambigProjID | tagAmbig(tags_for_ambiguities) | Ambig tag
-- |  ❌ ambigProjectID change to ambigProjID | projAmbig(project_ambiguities_for_tag_project) | Ambig tag
-- | ❌ motusProjectID change to tagProjID | batches(batches_for_tag_project,batches_for_receiver) | Tag (hit)
-- | ❌ motusProjID change to tagProjID | filters (Created locally) | Tag (hit)
-- | ❌ projectID1 change to tagProjID | projAmbig (API) | Tag (Ambig?)
-- | ❌ tagDepProjectID change to tagProjID | projBatch (Created locally) | Tag (hit)


<h3 dir="auto">Question</h3>
</body>
</html>

 1.  Why do some have motus prepended (e.g., motusProjectID, motusProjID)? Couldn't those remain projectID?
 
CBJ: No reason. You're right. Too wordy. But we may be able to keep consistency by assigning project ID at the tag and receiver (perhaps we should consider station instead? Another big change across the R package that should be tackled in a separate GIT. )

2.    Why is there a tagDepProjectID? Could this not be projectID or tagProjID if required?

CBJ: Yep, also bad. Change to tagProjID. Tags can only ever be in one project. 

3. Although we may not want to do this, would it not be better to have:
        recvProjID in Views
        tagProjID  in Views
        ambigProjID  in Tables/API
        projectID in Tables/API

CBJ: This is also another very good approach. Though @ajmacdonald3's suggestion maximizes consistency.

**Comment by denislepage** [comment]
Given that this is just affecting the views (yes?), one option to facilitate the transition would be to have a new view (distinct name) and keep the old one (that or rename to old one and create a new one with the old name).

One benefit of a new name is that people who already have databases could continue to use the old view (assuming we don't delete it), and new databases would only have the new version. Existing databases would also have the new one that they can transition to as well.

**Comment by cjardine-bsc** [comment]
#smart

**Comment by steffilazerte** [comment]
> **Paraphrased from @cjardine-bsc  and @ajmacdonald3**
> Everything should become `recvProjID`, `tagProjID` or `ambigProjID`

That makes a lot of sense and will clarify things immensely!

(For the other comments, which apply more generally to changing variable names, I've opened a [new issue](#1307) for that discussion)
